lib/rust/frp/src/io/timer/timeout.rs

```diff
@@ -0,0 +1,102 @@
+//! FRP bindings for browser `setTimeout` timer.
+
+use crate::prelude::*;
+pub use enso_web::traits::*;
+
+use crate as frp;
+
+use enso_web::window;
+use enso_web::Closure;
+
+
+// ================
+// === Timeout ===
+// ================
+
+/// Closure type alias for use in `setTimeout` call.
+type TimerClosure = Closure<dyn FnMut()>;
+
+/// One-shot timer.
+///
+/// The timer can be started or cancelled at any time using `start` and `cancel` inputs. After it is
+/// started, an `on_expired` event will be emitted at most once after the specified time provided
+/// during start. If the timer is cancelled before it expires, no event will be emitted during that
+/// run. Starting the timer again while it is already running is treated as a restart - previous run
+/// is canceled.
+///
+/// in `start`:       5ms--------20ms--3ms-------3ms-------
+/// in `cancel`:      ----------------------------x----x---
+/// out `on_expired`: -----x--------------x----------------
+///
+/// The timer is based on `setTimeout` browser API. That means there is no guarantee about the
+/// exact time the event will be emitted. It might be delayed if the browser event loop is busy. If
+/// you need to fire events periodically, prefer [`Interval`] timer in order to avoid unpredictable
+/// event trigger rate.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Timeout {
+    /// Starts the timer immediately with provided timeout value, specified in integer
+    /// milliseconds. If the timer was already started, it is cancelled and restarted.
+    pub start:      frp::Any<i32>,
+    /// Stops the timer if it is running. No `on_expired` events will be emitted until it is
+    /// started again.
+    pub cancel:     frp::Any,
+    /// Emitted when timer expires. At most one event is emitted for each
+    /// start of the timer.
+    pub on_expired: frp::Stream<()>,
+    raw_timeout:    Rc<RawTimeout>,
+}
+
+impl Timeout {
+    /// Constructor. Timer is initially not started.
+    pub fn new(network: &frp::Network) -> Self {
+        frp::extend! { network
+            on_expired <- any_mut();
+        }
+
+        let closure: TimerClosure = Closure::new(f!(() on_expired.emit(())));
+        let raw_timeout = Rc::new(RawTimeout::new(closure));
+
+        frp::extend! { network
+            start  <- any_mut::<i32>();
+            cancel <- any_mut::<()>();
+            eval   start((timeout) raw_timeout.start(*timeout));
+            eval_  cancel(raw_timeout.cancel());
+        }
+        let on_expired = on_expired.into();
+        Self { on_expired, start, cancel, raw_timeout }
+    }
+}
+
+#[derive(Debug)]
+struct RawTimeout {
```

Comment:
section missing

Reply:
Same thing as with `RawInterval` - I think this belongs to the `Timeout` section, as it is a critical implementation detail and part of the same concept.

Reply:
Lets do the section here as well, please.